drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,254 @@
+		case LED_COLOR_ID_GREEN:
+		case LED_COLOR_ID_BLUE:
+			break;
+		default:
+			LOG_ERR("%s: invalid channel to color mapping."
+				"Check the color-mapping DT property",
+				dev->name);
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static size_t ws2812_strip_length(const struct device *dev)
+{
+	const struct ws2812_uart_cfg *cfg = dev->config;
+
+	return cfg->length;
+}
+
+static const struct led_strip_driver_api ws2812_uart_api = {
+	.update_rgb = ws2812_strip_update_rgb,
+	.length = ws2812_strip_length,
+};
+
+#define WS2812_DEFAULT_RESET_DELAY_MS (50)
+#define WS2812_UART_NUM_PIXELS(idx)   (DT_INST_PROP(idx, chain_length))
+#define WS2812_UART_BUFSZ(idx)        (WS2812_NUM_COLORS(idx) * 8 * WS2812_UART_NUM_PIXELS(idx))
+
+/*
+ * Retrieve the channel to color mapping (e.g. RGB, BGR, GRB, ...) from the
+ * "color-mapping" DT property.
+ */
+#define WS2812_COLOR_MAPPING(idx)                                                                  \
+	static const uint8_t ws2812_uart_##idx##_color_mapping[] = DT_INST_PROP(idx, color_mapping)
+
+#define WS2812_UART_LUT(idx)                                                                       \
+	static const uint8_t ws2812_uart_##idx##_lut[] = DT_INST_PROP(idx, lookup_table)
+
+#define WS2812_NUM_COLORS(idx) (DT_INST_PROP_LEN(idx, color_mapping))
+
+#define WS2812_UART_LUT_LEN(idx) (DT_INST_PROP_LEN(idx, lookup_table))
+
+/* Get the latch/reset delay from the "reset-delay" DT property. */
+#define WS2812_RESET_DELAY(idx) DT_INST_PROP_OR(idx, reset_delay, WS2812_DEFAULT_RESET_DELAY_MS)
+
+#define UART_NODE(idx) DT_PARENT(DT_INST(idx, worldsemi_ws2812_uart))
+
+#define BITS_PER_RGB_BIT(idx) DT_INST_PROP(idx, bits_per_rgb_bit)
+
+#define UART_DATA_BITS(idx) DT_PROP(UART_NODE(idx), data_bits)
+
+#define WS2812_UART_DEVICE(idx)                                                                    \
+                                                                                                   \
+	BUILD_ASSERT(DT_PROP(UART_NODE(idx), tx_invert),                                           \
+		     "This driver depends on tx-invert UART property and "                         \
+		     "is not set. Please check your device-tree settings");                        \
+                                                                                                   \
+	BUILD_ASSERT(0 == (((UART_DATA_BITS(idx) + 2) % BITS_PER_RGB_BIT(idx))),                   \
+		     "UART data bits+2 should be multiple of bits_per_rgb_bit"                     \
+		     "property.Please check your device-tree settings");                           \
+                                                                                                   \
+	BUILD_ASSERT(WS2812_UART_LUT_LEN(idx) ==                                                   \
+			     1 << ((UART_DATA_BITS(idx) + 2) / BITS_PER_RGB_BIT(idx)),             \
+		     "UART lookup_table property's length != 2^((data-bits+2)/bits-per-rgb-bit)"   \
```

Comment:
Honestly a such error message won't make any sense to reader (even to you in a couple of months) :)